### DISCUSSION TITLE
MicroPython Libraries for Air Quality Sensors

### DISCUSSION THREAD

**Discussion by ayenesams** [micropython/micropython#11409] [Libraries & Drivers] [answered]
Newbie here. I am currently working on a capstone project that involves integrating various sensors into a MicroPython-based system while using ESP32 as the microcontroller. Specifically, I am looking for resources, libraries, or code examples related to the following sensors:
1. SPS30
2. ULPSM-SO2 968-006
3. MiCS-4514
4. MiCS-VZ-89TE

Thanks!

**Comment by mcauser** [comment, marked as answer]
You can pretty much just google "device-name micropython" to get hits.

SPS30
* https://github.com/dvsu/sps30
* https://github.com/szajakubiak/Python-Sensirion-SPS30/blob/master/sps30.py

ULPSM-SO2 968-006
* not much

MiCS-4514
* https://github.com/BuysDB/MICS-Python

MiCS-VZ-89TE
* https://github.com/papamidas/esp32_co2sensor

Drivers for other air quality sensors can be found here:
* https://awesome-micropython.com/#air-quality
